reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java

```diff
@@ -88,16 +89,38 @@
 
 	@Override
 	public void doFirst() {
-		ContextSnapshot contextSnapshot = ContextSnapshot.capture(this.originalContext);
-
-		try (ContextSnapshot.Scope ignored = contextSnapshot.setThreadLocalValues()) {
-			this.scope = this.subscribeToTerminalObservation
-				.start()
-				.openScope();
-			//reacquire the scope from ThreadLocal
-			//tap context hasn't been initialized yet, so addToContext can now use the Scope
-			ContextSnapshot contextSnapshot2 = ContextSnapshot.capture(this.originalContext);
-			this.contextWithScope = contextSnapshot2.updateContext(Context.of(this.originalContext));
+		if (Micrometer.isContextPropagationAvailable()) {
+			//load any parent from either the Reactor context or if none from the ThreadLocal
+
+			ContextSnapshot contextSnapshot = ContextSnapshot.captureUsing(Micrometer.OBSERVATION_CONTEXT_PREDICATE, this.originalContext);
+
+			try (ContextSnapshot.Scope ignored = contextSnapshot.setThreadLocalValues(Micrometer.OBSERVATION_CONTEXT_PREDICATE)) {
+				this.scope = this.subscribeToTerminalObservation
+					.start()
+					.openScope();
+				//with the new scope comes a new Observation which we can put in the Context for inner publishers
+				Observation newObservation = this.configuration.registry.getCurrentObservation();
+				this.contextWithScope = Context.of(this.originalContext)
+					.put(Micrometer.OBSERVATION_CONTEXT_KEY, newObservation);
+			}
```

Comment:
Just double checking, is it really necessary to create a `ThreadLocal` for the new `Observation` or can it be more like [what was done](https://github.com/reactor/reactor-netty/blob/db75dba2afec4a194091d11c902f3d71c888a4ed/reactor-netty-core/src/main/java/reactor/netty/channel/MicrometerChannelMetricsHandler.java#L125) in Reactor Netty.